app/services/cocina_generator/structural/generator.rb

```diff
@@ -27,36 +27,34 @@ def generate
       # 2) Rebuild filesets that have attached_files that are preserved
       # 3) Add new filesets that have attached_files in the local store.
       def build_filesets
-        filesets = find_preserved_filesets.to_h do |filename, fileset|
+        (work_version.staged_files.map { |af| build_fileset(attached_file: af) } + find_preserved_filesets.values)
+          .sort_by { |file_set_hash| file_set_hash.dig(:structural, :contains, 0).filename }
+      end
+
+      # @return [Hash] map of filename to matching fileset's cocina for preserved files.
+      def find_preserved_filesets # rubocop:disable Metrics/AbcSize
+        return {} unless cocina_obj
+
+        # h2 only has one file per fileset
+        cocina_obj.structural.contains
+                  .select { |fileset| preserved_file_names.include?(fileset.structural.contains.first.filename) }
+                  .to_h do |fileset|
+          filename = fileset.structural.contains.first.filename
           [
             filename,
             build_fileset(attached_file: get_preserved_file(filename), fileset:)
           ]
         end
-        filesets.values + work_version.staged_files.map { |af| build_fileset(attached_file: af) }
       end
 
-      # @return [Hash] map of filename to matching fileset's cocina for preserved files.
-      def find_preserved_filesets
-        return {} unless cocina_obj
-
-        preserved_file_names = work_version.preserved_files.map(&:filename).map(&:to_s)
-        create_filesmap(preserved_file_names)
+      def preserved_file_names
+        work_version.preserved_files.map(&:filename).map(&:to_s)
       end
 
       def get_preserved_file(filename)
         work_version.preserved_files.find { |attached_file| attached_file.filename.to_s == filename }
       end
 
-      # rubocop:disable Rails/IndexBy
-      def create_filesmap(preserved_file_names)
-        # h2 only has one file per fileset
-        cocina_obj.structural.contains
-                  .select { |fileset| preserved_file_names.include?(fileset.structural.contains.first.filename) }
-                  .to_h { |fileset| [fileset.structural.contains.first.filename, fileset] }
-      end
-      # rubocop:enable Rails/IndexBy
```

Comment:
I inlined this above since it was only used in one spot and added more clarity to what `find_preserved_filesets` was/is doing.